### PR TITLE
TVIST1-754: Added finishedOn property to cases and logic for setting …

### PR DESCRIPTION
**Pull request by jekuaitk**
…this

https://jira.itkdev.dk/browse/TVIST1-754

* Adds `finishedOn` property to cases.
* Adds logic for setting `finishedOn` when reaching final status.

### CODE REVIEW

File: src/Entity/Board.php [hunk 1]
```diff
@@ -451,4 +451,14 @@ public function setHearingPostResponseTemplate(?MailTemplate $hearingPostRespons
 
         return $this;
     }
+
+    public function getFinalStatus(): string
+    {
+        $rawPlaces = explode(
+            PHP_EOL,
+            trim($this->getStatuses())
+        );
```
Comment (rimi-itk) [inline src/Entity/Board.php:460]: Use 

```php
        $rawPlaces = array_filter(
            array_map(
                'trim',
                explode(
                    PHP_EOL,
                    $this->getStatuses()
                )
            )
        );
```

to make this work as expected, i.e. to trim all lines after exploding and retain only non-empty lines.

File: src/Entity/Board.php [hunk 2]
```diff
@@ -451,4 +451,14 @@ public function setHearingPostResponseTemplate(?MailTemplate $hearingPostRespons
...
+
+        return trim(end($rawPlaces));
```
Comment (rimi-itk) [inline src/Entity/Board.php:462]: `end` returns `false` if the array is empty (cf. https://www.php.net/manual/en/function.end.php#refsect1-function.end-returnvalues). You should probably handle this somehow, e.g. with

```php
return end($rawPlaces) ?: '';
```

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: src/Entity/Board.php (hunk 1 above)
Comment (jekuaitk) [reply src/Entity/Board.php:460]: 👍

File: src/Entity/Board.php (hunk 2 above)
Comment (jekuaitk) [reply src/Entity/Board.php]: 👍

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**